### PR TITLE
feat(parental-leave): Allow to start on estimated or actual date of birth

### PR DESCRIPTION
**Pull request by madebynoam**
Change the first period start options to the following:

* Estimated date of birth
* Actual date of birth
* Specific date

Also refactor options into an enum.

## Why
https://app.asana.com/0/1182378413629561/1200111966013809/f

## Screenshots / Gifs

image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/parental-leave/src/constants.ts [hunk 1]
```diff
@@ -6,3 +6,9 @@ export enum API_MODULE_ACTIONS {
   assignEmployer = 'assignEmployer',
   sendApplication = 'sendApplication',
 }
+
+export enum StartDateOptions {
+  ESTIMATED_DATE_OF_BIRTH = 'estimatedDateOfBirth',
+  DATE_OF_BIRTH = 'dateOfBirth', // The actual date of birth
```
Comment (jeremybarbet) [inline libs/application/templates/parental-leave/src/constants.ts:12]: I would rename it to make it explicit and to not need the comment

```suggestion
  ACTUAL_DATE_OF_BIRTH = 'actualDateOfBirth',
```

File: libs/application/templates/parental-leave/src/fields/FirstPeriodStart/index.tsx [hunk 2]
```diff
@@ -9,8 +9,13 @@ import { useFormContext } from 'react-hook-form'
 import { getExpectedDateOfBirth } from '../../parentalLeaveUtils'
 import { parentalLeaveFormMessages } from '../../lib/messages'
 import { useLocale } from '@island.is/localization'
+import { StartDateOptions } from '../../constants'
 
-type ValidAnswers = 'dateOfBirth' | 'specificDate' | undefined
+type ValidAnswers =
```
Comment (jeremybarbet) [inline libs/application/templates/parental-leave/src/fields/FirstPeriodStart/index.tsx:14]: You can also write 

```typescript
type ValidAnswers = StartDateOptions | undefined;
```

**Review by eirikurn — APPROVED** [review]

File: libs/application/templates/parental-leave/src/lib/messages.ts [hunk 3]
```diff
@@ -600,20 +600,18 @@ export const parentalLeaveFormMessages: MessageDir = {
     },
     description: {
       id: 'pl.application:periods.first.period.description',
-      defaultMessage: `Þú mátt kjósa að byrja á áætluðum fæðingardegi, eða ákveðinni dagsetningu. Athugaðu að ekki er hægt að nýta réttindi til fæðingarorlofs 18 mánuðum eftir fæðingu barnsins.`,
-      description: `You can choose to start on the date of birth, or on a specific date. Please note, that your rights end 18 months after the date of birth.`,
+      defaultMessage: `Athugaðu að ekki er hægt að nýta réttindi til fæðingarorlofs 18 mánuðum eftir fæðingu barnsins.`,
+      description: `Please note, that your rights end 18 months after the date of birth.`,
```
Comment (jeremybarbet) [inline libs/application/templates/parental-leave/src/lib/messages.ts:604]: Did you also update the contentful's namespace at the same time? and run the extract-strings script to update the messages?

The description is not used for the English translation so it won't update it automatically on Contentful. It's also better to replace with some information's copy so the content editor has a better understanding of where/what is the message used for.

**Review by jeremybarbet — CHANGES_REQUESTED** [review]
Looks good overall, just some minor comments 👍

File: libs/application/templates/parental-leave/src/constants.ts (hunk 1 above)
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/constants.ts]: Good point, should be updated now.

File: libs/application/templates/parental-leave/src/fields/FirstPeriodStart/index.tsx (hunk 2 above)
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/fields/FirstPeriodStart/index.tsx:14]: Nice! changed it now

File: libs/application/templates/parental-leave/src/lib/messages.ts (hunk 3 above)
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/lib/messages.ts:604]: Thanks for this info. 
I updated the descriptions and ran `yarn nx run application-templates-parental-leave:extract-strings` now. 

Do I need to do anything else for this?

**Comment by madebynoam** [comment]
Thanks for the review! Should be ready again now.

**Review by jeremybarbet — APPROVED** [review]
LGTM 🚀

**Pull request merged**